codegen/smithy-kotlin-codegen/src/main/kotlin/software/amazon/smithy/kotlin/codegen/rendering/waiters/KotlinJmespathExpressionVisitor.kt

```diff
@@ -107,26 +103,29 @@ class KotlinJmespathExpressionVisitor(
 
     private fun subfield(expression: FieldExpression, parentName: String): VisitedExpression {
         val member = currentShape.targetOrSelf(ctx.model).getMember(expression.name).getOrNull()
-            ?: throw CodegenException("reference to nonexistent member '${expression.name}' of shape $currentShape")
 
         val name = expression.name.toCamelCase()
-        val nameExpr = ensureNullGuard(currentShape, name)
-
-        val memberTarget = ctx.model.expectShape(member.target)
-        val unwrapExpr = when {
-            memberTarget.isEnum -> "value"
-            memberTarget.isEnumList -> "map { it.value }"
-            memberTarget.isEnumMap -> "mapValues { (_, v) -> v.value }"
-            memberTarget.isBlobShape || memberTarget.isTimestampShape ->
-                throw CodegenException("acceptor behavior for shape type ${memberTarget.type} is undefined")
-            else -> null
+        val nameExpr = addNullGuard(currentShape, name)
+        var unwrapExpr: String? = null
+
+        if (member != null) {
+            val memberTarget = ctx.model.expectShape(member.target)
+            unwrapExpr = when {
+                memberTarget.isEnum -> "value"
+                memberTarget.isEnumList -> "map { it.value }"
+                memberTarget.isEnumMap -> "mapValues { (_, v) -> v.value }"
+                memberTarget.isBlobShape || memberTarget.isTimestampShape ->
+                    throw CodegenException("acceptor behavior for shape type ${memberTarget.type} is undefined")
+                else -> null
+            }
         }
+
         val codegen = buildString {
             append("$parentName$nameExpr")
-            unwrapExpr?.let { append(ensureNullGuard(member, it)) }
+            unwrapExpr?.let { append(addNullGuard(member, it)) }
         }
 
-        shapeCursor.addLast(member)
+        if (member != null) shapeCursor.addLast(member)
```

Comment:
Same comment here about using the Kotlin null helper methods: `member?.let { shapeCursor.addLast(it) }`